src/apps/relay/netengine.c

```diff
@@ -785,35 +789,36 @@ static void handle_relay_auth_message(struct relay_server *rs, struct auth_messa
 }
 
 static void relay_receive_message(struct bufferevent *bev, void *ptr) {
-  struct message_to_relay sm;
+  struct message_to_relay sm = {0};
   int n = 0;
   struct evbuffer *input = bufferevent_get_input(bev);
   struct relay_server *rs = (struct relay_server *)ptr;
 
-  while ((n = evbuffer_remove(input, &sm, sizeof(struct message_to_relay))) > 0) {
-
-    if (n != sizeof(struct message_to_relay)) {
-      perror("Weird buffer error\n");
-      continue;
-    }
+  n = evbuffer_get_length(input);
+  if (n < sizeof(struct message_to_relay)) {
```

Comment:
wouldn't this be better as `sizeof(sm)` to ensure a mistake isn't made if `sm` is changed to a different variable type in the future?